### PR TITLE
Add Revive Linter

### PR DESCRIPTION
**Pull request by cpheps**
## Description of Changes
- Removed golangci-lint in favor of revive linter
- Added github action to run linter
- Fixed linter issues

## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

File: revive/config.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+ignoreGeneratedHeader = false
+severity = "warning"
+confidence = 0.8
+errorCode = 0
+warningCode = 0
```
Comment (BinaryFissionGames) [inline revive/config.toml:5]: Will this fail CI if we violate the linter? I don't know if I really understand what errorCode and warningCode configure, so maybe these don't effect the action.

Comment (cpheps) [reply revive/config.toml:5]: Let me double check. I copied this from their proposed config but forgot to follow up on it.

File: operator/builtin/input/file/rotation_test.go [hunk 2]
```diff
@@ -17,8 +17,11 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+//revive:disable:var-naming
 const WINDOWS_OS = "windows"
 
+//revive:enable:var-naming
```
[the remark this answers is not in the document]
Comment (cpheps) [reply operator/builtin/input/file/rotation_test.go:23]: Let me double check this.

File: revive/config.toml (hunk 1 above)
Comment (cpheps) [reply revive/config.toml:5]: Good catch on these I changed them to return `1`.

**Review by BinaryFissionGames — APPROVED** [review]

**Review by dehaansa — APPROVED** [review]

**Pull request merged**